mne/io/egi/egimff.py

```diff
@@ -500,6 +521,22 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         stim_in = idx[stim_out] - bounds[1]
         pns_out = np.where((idx >= bounds[2]) & (idx < bounds[3]))[0]
         pns_in = idx[pns_out] - bounds[2]
+        eeg_out = eeg_out[:, np.newaxis]
+        pns_out = pns_out[:, np.newaxis]
+
+        # take into account events (already extended to correct size)
+        data[stim_out, :] = egi_info['egi_events'][stim_in, start:stop]
+        del stim_out
+
+        # Convert start and stop to limits in terms of the data
+        # actually on disk, plus an indexer (disk_use_idx) that populates
+        # the potentially larger `data` with it, taking skips into account
+        disk_samps = egi_info['disk_samps'][start:stop]
+        disk_use_idx = np.where(disk_samps > -1)[0]
+        if len(disk_use_idx):
```

Comment:
what happens otherwise? I mean: if there are only `-1` in the `start:stop` segment? There is no need to open and read the data in such case, but the file is still opened later in the code `with open(...) as fid:`.

Reply:
yes this will happen but it should be a negligible performance cost. I'd rather not complicate the code by special casing the opening it unless we see some performance loss

Reply:
sure, sounds ok!